tracing-subscriber/src/registry/mod.rs

```diff
@@ -238,17 +348,106 @@ where
         })
     }
 
+    /// Returns an iterator over all parents of this span, starting with this span,
+    /// ordered from leaf to root.
+    ///
+    /// The iterator will first return the span, then the span's immediate parent,
+    /// followed by that span's parent, and so on, until it reaches a root span.
+    ///
+    /// ```rust
+    /// use tracing::{span, Subscriber};
+    /// use tracing_subscriber::{
+    ///     layer::{Context, Layer},
+    ///     prelude::*,
+    ///     registry::{LookupSpan, Registry},
+    /// };
+    /// # // Use static mut to avoid tainting the visible part of the example with state
+    /// # static mut last_entered_scope: Vec<&'static str> = Vec::new();
```

Comment:
i really don't think it's worth adding unsafe code here just to make an assertion about the example that isn't even visible to the reader of the docs. IMO, doctests are not really intended to be used for actual testing, the reason they are compiled and run is to ensure the code in the example compiles and is kept up to date. it's nice to be able to also check that simple assertions visible to the reader are correct, but i don't think it's important enough to justify adding unsafe code --- especially since it will now show up in auditing tools when this crate previously had no unsafe code.

why don't we move the assertions out of the examples and into an actual test, and use a mutex in the test rather than a `static mut` (even though it might not be  _strictly_ necessary)?

Reply:
Sure, fair enough.

Reply:
b651701c0931606399e1243029169591a9e04bf2